packages/block-editor/src/components/inserter/block-patterns-tab.js

```diff
@@ -7,20 +7,21 @@ import { fromPairs } from 'lodash';
  * WordPress dependencies
  */
 import { useMemo, useCallback } from '@wordpress/element';
-import { __, _x } from '@wordpress/i18n';
+import { __ } from '@wordpress/i18n';
 import { useAsyncList } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import InserterPanel from './panel';
+import PatternInserterPanel from './pattern-panel';
 import { searchItems } from './search-items';
 import InserterNoResults from './no-results';
 import usePatternsState from './hooks/use-patterns-state';
 import BlockPatternList from '../block-patterns-list';
 
 function BlockPatternsSearchResults( { filterValue, onInsert } ) {
-	const [ patterns, , onClick ] = usePatternsState( onInsert );
+	const [ , patterns, , onClick ] = usePatternsState( onInsert, 'all' );
```

Comment:
I wonder if we should return an object from usePatternsState instead of an array as this api is a bit hard use/understand - an object return would make it simpler `const { patterns, onClick } = usePatternsState( onInsert, 'all' );`

Reply:
Yeah, that would make more sense to me.

Reply:
Yep makes sense. It was already this way for some reason, but I don't see any reason why it needs to be.

Reply:
The original idea is that if follows the `useState` pattern. The advantage of arrays is that it allows the consumer to name the arguments as you want to avoid duplicated variables...

If we were to changes this, we should change it for useBlockTypesState too as these two are very similar.